Formula/libswiftnav.rb

```diff
@@ -1,9 +1,15 @@
 class Libswiftnav < Formula
   desc "C library implementing GNSS related functions and algorithms"
-  homepage "https://github.com/swift-nav/libswiftnav-legacy"
-  url "https://github.com/swift-nav/libswiftnav-legacy/archive/v0.21.tar.gz"
-  sha256 "087c7264c0d0d735414f8bffbfa52ab44696c500ba14a43262d98d6aa093221f"
-  license "LGPL-3.0"
+  homepage "https://github.com/swift-nav/libswiftnav"
+  url "https://github.com/swift-nav/libswiftnav.git",
+    tag:      "v2.4.2",
+    revision: "ed4a6b0564ead39e5089b5f97ab93ab9b4009800"
```

Comment:
```suggestion
      tag:      "v2.4.2",
      revision: "ed4a6b0564ead39e5089b5f97ab93ab9b4009800"
```
Let's align the arguments to `url`. (Most other formulae do this too.) Can we not use a tarball?

Reply:
Checking out the Git repository is necessary, unfortunately. `/cmake/common` is required to build but it's a submodule in the Git repository (pointing to [swift-nav/cmake](https://github.com/swift-nav/cmake)). The tag archive files simply have an empty `/cmake/common` directory, so building fails.

Reply:
We could install it as a resource. It'll be a bit more annoying to maintain, but we might be better off as we no longer do shallow Git clones. Tarballs are better for caching too.

Reply:
It's possible and it also builds faster that way (37 seconds with Git, 8 seconds with tarballs). The installed files end up the same, so it's mostly a question of whether ease of maintenance outweighs any benefits of using a tarball and a `resource` block for `cmake`.

Using a `resource` block would make it more challenging to update the formula correctly. Contributors/maintainers would need to manually check the `libswiftnav` repository's `cmake` directory using the same version tag (e.g., https://github.com/swift-nav/libswiftnav/tree/v2.4.2/cmake) to identify the referenced commit hash to use in the `resource` block.

Here's the diff compared to the current commit in this PR (it seems to build without `pkg-config` either way):

```diff
diff --git a/Formula/libswiftnav.rb b/Formula/libswiftnav.rb
index dbb0e1678ab54a40c19426d6652d25f1c26b5a89..14ec9ca6e7c404ef6ec1c3ce09be40d45246ac24 100644
--- a/Formula/libswiftnav.rb
+++ b/Formula/libswiftnav.rb
@@ -1,9 +1,8 @@
 class Libswiftnav < Formula
   desc "C library implementing GNSS related functions and algorithms"
   homepage "https://github.com/swift-nav/libswiftnav"
-  url "https://github.com/swift-nav/libswiftnav.git",
-      tag:      "v2.4.2",
-      revision: "ed4a6b0564ead39e5089b5f97ab93ab9b4009800"
+  url "https://github.com/swift-nav/libswiftnav/archive/v2.4.2.tar.gz"
+  sha256 "9dfe4ce4b4da28ffdb71acad261eef4dd98ad79daee4c1776e93b6f1765fccfa"
   license "LGPL-3.0-only"
 
   livecheck do
@@ -23,9 +22,18 @@ class Libswiftnav < Formula
   end
 
   depends_on "cmake" => :build
-  depends_on "pkg-config" => :build
+
+  # Check the `/cmake` directory for a given version tag
+  # (e.g., https://github.com/swift-nav/libswiftnav/tree/v2.4.2/cmake)
+  # to identify the referenced commit hash in the swift-nav/cmake repository.
+  resource "swift-nav/cmake" do
+    url "https://github.com/swift-nav/cmake/archive/fd8c86b87d2b18261691ef8db1f6fd9906911b82.tar.gz"
+    sha256 "7b6995bcc97d001cfe5c4741a8fa3637bc4dc2c3460b908585aef5e7af268798"
+  end
 
   def install
+    (buildpath/"cmake/common").install resource("swift-nav/cmake")
+
     mkdir "build" do
       system "cmake", "..", *std_cmake_args
       system "make", "install"
```

Reply:
Let's try the tarball. If it gets too annoying to maintain, we can switch back to a Git checkout.